### DISCUSSION TITLE
macOS  Python 2.7 – The Lights Are Almost Out

### DISCUSSION THREAD

**Discussion by OldSubSailor** [Ortho4XP/Ortho4XP#7] [General]
I think that implicit to this is that there will be NO version of Python included with macOS when they finally do eliminate Python 2, and the same for Ruby.  Therefore we will need to ensure that Python is installed in the instructions for macOS. 

The first change to be aware of, and not before time many will say, is that Apple are finally pushing people (and themselves) off Python 2.

In Monterey, an app that uses Python 2.7 now triggers a UI alert indicating that the developer must update the app to ensure it will work in future versions of macOS. We saw this same general ‘scare’ tactic work quite successfully when Apple wanted to push developers away from kexts in macOS Catalina.

Throwing alerts at users and having users moan at developers worked very well in that case, and we’re sure it will do in this. There will inevitably be some complaints from both users and devs, but really, Python 2.7 has stayed up well past its bedtime.

www.sentinelone.com

**Comment by ARobinsonAtTTC** [comment]
Honestly this is probably a good thing. We already need to make sure that Python3 is installed for Ortho4XP to work. (Python3 is not backwards compatible with Python2). 

Python2 is no longer supported by the Python Software Foundation and hasn't been for at least a year now.
